File: 01realTimeComputeAddressOrderAmount/src/main/java/com/hhh/temp/tools/RandomTools.java
package com.hhh.temp.tools;

import java.util.Random;

/**
 * Created by huanghh on 2017/5/16.
 */
public class RandomTools {
    public static int getRandomI(int maxI) {
        int randomI = getRandomI(0,maxI);
        return randomI;
    }
    public static int getRandomI(int minI,int maxI) {
        if (minI > maxI) {
            maxI = minI;
            minI = 0;
        }

        Random random = new Random();

        int randomI = random.nextInt(maxI);
        if (randomI < minI) {
            randomI = randomI + minI;
        }
        return randomI;
    }
}

File: 02realTimeComputeAddressOrderAmountByScala/src/main/scala/com/hhh/temp/tools/RandomTools.scala
package com.hhh.temp.tools

import java.util.Random

/**
  * Created by huanghh on 2017/5/16.
  */
object RandomTools {
  def getRandomI(maxI: Int): Int = {
    val randomI: Int = getRandomI(0, maxI)
    randomI
  }

  def getRandomI(minI: Int, maxI: Int): Int = {
    var _minI: Int = minI
    var _maxI: Int = maxI
    if (_minI > _maxI) {
      _maxI = minI
      _minI = 0
    }
    val random: Random = new Random
    var randomI: Int = random.nextInt(_maxI)
    if (randomI < _minI) randomI = randomI + _minI
    randomI
  }
}
